tests/testsuite/cargo_information/within_workspace/stderr.log

```diff
@@ -1,3 +1,4 @@
     Updating `dummy-registry` index
  Downloading crates ...
   Downloaded my-package v0.1.1+my-package (registry `dummy-registry`)
+    Updating crates.io index
```

Comment:
I'm unsure if this makes sense, but it appears to be a bug. My understanding is that we should throw an error if users specify an index or registry while picking the source ID from the lock file.

Reply:
We need `-registry=dummy-registry` for this case because `dummy-registry` is a replacement for `crates.io`.

Reply:
https://github.com/hi-rustin/cargo/blob/rustin-patch-cargo-test-support/src/cargo/sources/registry/remote.rs#L368

I think the `dummy-registry` came from here.

Reply:
Oh, I get the reason. This caused by https://github.com/hi-rustin/cargo-information/blob/cb89385af75c4ed0685b8a3dd763f46322e6fa6c/src/ops/info.rs#L152

I suppose we need to use `replacement_sid` here. I don't know why Cargo always uses `builtin_replacement_sid` here.